Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean

```diff
@@ -222,6 +222,18 @@ theorem det_submatrix_equiv_self (e : n ≃ m) (A : Matrix m m R) :
   intro i
   rw [Equiv.permCongr_apply, Equiv.symm_apply_apply, submatrix_apply]
 
+/-- Permuting rows and columns with two equivalences does not change the absolute value of the
+determinant. -/
+@[simp]
+theorem abs_det_submatrix_equiv_equiv {R : Type*} [LinearOrderedCommRing R]
+    (e₁ e₂ : n ≃ m) (A : Matrix m m R) :
+    |(A.submatrix e₁ e₂).det| = |A.det| := by
+  have hee : e₂ = e₁.trans (e₁.symm.trans e₂) := by ext; simp
+  rw [hee]
+  show |((A.submatrix id (e₁.symm.trans e₂)).submatrix e₁ e₁).det| = |A.det|
+  rw [Matrix.det_submatrix_equiv_self, Matrix.det_permute']
+  cases' Int.units_eq_one_or (sign (e₁.symm.trans e₂)) with he he <;> rw [he] <;> simp
```

Comment:
I would have hoped that we could do this as one rewrite, by rewriting with `abs_mul` and then saying the cast of `Units Int` always has absolute value 1. I did find `AbsoluteValue.map_units_intCast`, but that's for bundled absolute value operators.

Could you extract a lemma saying `|((u : Int) : R)| = 1` for `u : Units Int`?

Reply:
Do you want to eliminate line 234 as well, or just simplify line 235 by this change?

Reply:
I mean that we could make line 234 into a oneliner along the lines of:
```lean
  rw [Matrix.det_submatrix_equiv_self, Matrix.det_permute', abs_mul, abs_units_intCast, one_mul]
```